test/integration/nighthawk_grpc_service.py

```diff
@@ -1,50 +1,63 @@
-import json
 import logging
-import os
 import socket
 import subprocess
-import sys
 import tempfile
 import threading
 import time
-from string import Template
 
 from common import IpVersion
 
 
+# TODO(oschaaf): unify some of this code with the test server wrapper.
 class NighthawkGrpcService(object):
   """
-    Base class for running a server in a separate process.
-    """
+  Class for running the Nighthawk gRPC service in a separate process.
+  Usage:
+    grpc_service = NighthawkGrpcService("/path/to/nighthawk_service"), "127.0.0.1", IpVersion.IPV4)
+    if grpc_service.start():
+      .... You can talk to the Nighthawk gRPC service at the 127.0.0.1:grpc_service.server_port ...
+
+  Attributes:
+  server_ip: IP address used by the gRPC service to listen. 
+  server_port: An integer, indicates the port used by the gRPC service to listen. 
```

Comment:
Looks like in the current design this would be a value of zero before start is called. That indicates that the object here can be in an "invalid" state. If we really wanted to fix it, we would make the  NighthawkGrpcService class private and add a public factory function that would both create the instance and call start on it. The factory function would then return the class instance in a valid state.

The other option is to at least document the behavior here where we are talking about the attribute. Indicating that it will be zero if the class wasn't used properly.

Reply:
Added a comment about that, and made sure to reset the port to 0 after stopping the server to be consistent: 75ed6acc89f5937e90bc9ef617c7e1b535d1e0ac.